hazelcast/src/main/java/com/hazelcast/config/HostVerification.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import java.util.Properties;
+
+/**
+ * TLS host verification configuration holder.
+ */
+public class HostVerification {
+
+    private String policyClassName;
```

Comment:
Is providing the configurable implementation necessary? I would try to avoid this duplication in configuration stuff until customers ask for it.

Reply:
> Is providing the configurable implementation necessary?

Not strictly necessary, but it would allow users to configure implementations which cannot have a public no-args constructor. Also, it's there in most Hazelcast config's I recall. Examples:
- `QuorumConfig`: `quorumFunctionClassName` & `quorumFunctionImplementation`
- `SSLConfig` (posted above)
- `ListenerConfig`: `className` & `implementation`
- `DiscoveryConfig`: `nodeFilter` & `nodeFilterClass`
etc